File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/LifeCycleFragment.java
package com.stedi.randomimagegenerator.app.view.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.CallSuper;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.stedi.randomimagegenerator.app.di.Components;
import com.stedi.randomimagegenerator.app.other.logger.Logger;

import javax.inject.Inject;

public abstract class LifeCycleFragment extends Fragment {
    private static final boolean LOG = true;

    @Inject Logger logger;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (LOG) {
            Components.getAppComponent(context).inject(this);
            logger.log(this, "onAttach");
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (LOG)
            logger.log(this, "onCreate");
    }

    @CallSuper
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (LOG)
            logger.log(this, "onCreateView");
        return null;
    }

    @CallSuper
    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        if (LOG)
            logger.log(this, "onViewCreated");
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (LOG)
            logger.log(this, "onActivityCreated");
    }

    @Override
    public void onStart() {
        super.onStart();
        if (LOG)
            logger.log(this, "onStart");
    }

    @Override
    public void onResume() {
        super.onResume();
        if (LOG)
            logger.log(this, "onResume");
    }

    @Override
    public void onPause() {
        super.onPause();
        if (LOG)
            logger.log(this, "onPause");
    }

    @CallSuper
    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        if (LOG)
            logger.log(this, "onSaveInstanceState");
    }

    @Override
    public void onStop() {
        super.onStop();
        if (LOG)
            logger.log(this, "onStop");
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (LOG)
            logger.log(this, "onDestroyView");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (LOG)
            logger.log(this, "onDestroy");
    }

    @Override
    public void onDetach() {
        super.onDetach();
        if (LOG)
            logger.log(this, "onDetach");
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/LifeCycleFragment.kt
package com.stedi.randomimagegenerator.app.view.fragments.base

import android.content.Context
import android.os.Bundle
import android.support.annotation.CallSuper
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import timber.log.Timber

abstract class LifeCycleFragment : Fragment() {
    private val LOG = false

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (LOG) Timber.i("onAttach")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (LOG) Timber.i("onCreate")
    }

    @CallSuper
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (LOG) Timber.i("onCreateView")
        return null
    }

    @CallSuper
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        if (LOG) Timber.i("onViewCreated")
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (LOG) Timber.i("onActivityCreated")
    }

    override fun onStart() {
        super.onStart()
        if (LOG) Timber.i("onStart")
    }

    override fun onResume() {
        super.onResume()
        if (LOG) Timber.i("onResume")
    }

    override fun onPause() {
        super.onPause()
        if (LOG) Timber.i("onPause")
    }

    @CallSuper
    override fun onSaveInstanceState(outState: Bundle) {
        if (LOG) Timber.i("onSaveInstanceState")
    }

    override fun onStop() {
        super.onStop()
        if (LOG) Timber.i("onStop")
    }

    override fun onDestroyView() {
        super.onDestroyView()
        if (LOG) Timber.i("onDestroyView")
    }

    override fun onDestroy() {
        super.onDestroy()
        if (LOG) Timber.i("onDestroy")
    }

    override fun onDetach() {
        super.onDetach()
        if (LOG) Timber.i("onDetach")
    }
}